sdk/notificationhubs/notification-hubs/src/api/clientContext.ts

```diff
@@ -69,46 +54,44 @@ export function createClientContext(
   return new NotificationHubsServiceClient(connectionString, hubName, options);
 }
 
-class NotificationHubsServiceClient extends ServiceClient implements NotificationHubsClientContext {
-  sasTokenProvider: SasTokenProvider;
+class NotificationHubsServiceClient implements NotificationHubsClientContext {
+  sasTokenCredential: SasTokenCredential;
   baseUrl: string;
   hubName: string;
+  client: Client;
+  httpClient: HttpClient;
 
   constructor(
     connectionString: string,
     hubName: string,
     options: NotificationHubsClientOptions = {}
   ) {
-    super({
-      deserializationOptions: {
-        parseXML,
-      },
-      serializationOptions: {
-        stringifyXML,
-      },
-      userAgentOptions: {
-        userAgentPrefix: `azsdk-js-messaging-notificationhubs/${constants.SDK_VERSION}`,
-      },
-      ...options,
-    } as InternalClientPipelineOptions);
-
     this.hubName = hubName;
 
     const parsedConnection = parseNotificationHubsConnectionString(connectionString);
-    this.baseUrl = parsedConnection.endpoint;
-    this.sasTokenProvider = createTokenProviderFromConnection(
+    // Node doesn't allow change in protocol but browsers do, so doing a string replace
+    this.baseUrl = parsedConnection.endpoint.replace("sb://", "https://");
+    this.sasTokenCredential = createTokenCredentialFromConnection(
       parsedConnection.sharedAccessKey,
       parsedConnection.sharedAccessKeyName
     );
+
+    this.httpClient = createDefaultHttpClient();
+    this.client = getClient(this.baseUrl, {
+      userAgentOptions: {
+        userAgentPrefix: `azsdk-js-messaging-notificationhubs/${constants.SDK_VERSION}`,
+      },
+      ...options,
+    });
   }
 
   async createHeaders(
     operationName: string,
     rawHeaders?: Record<string, string>
   ): Promise<HttpHeaders> {
-    const authorization = await this.sasTokenProvider.getToken(this.baseUrl);
+    const authorization = await this.sasTokenCredential.getToken(this.baseUrl);
     const headers = createHttpHeaders(rawHeaders);
-    headers.set("Authorization", authorization.token);
+    headers.set("Authorization", authorization!.token);
```

Comment:
Can we either handle the authorization being null or fix the method return?

Reply:
Fixed